waku/README.md

```diff
@@ -0,0 +1,301 @@
+| Topic | 4 bytes of arbitrary data |
+| Data | byte array of arbitrary size (contains encrypted message) |
+| Nonce |  8 bytes of arbitrary data (used for PoW calculation) |
+
+### Contents of Data Field of the Message (Optional)
+
+This section outlines the optional description of Data Field to set up an example. Later it may be moved to a separate EIP.
+
+It is only relevant if you want to decrypt the incoming message, but if you only want to send a message, any other format would be perfectly valid and must be forwarded to the peers.
+
+Data field contains encrypted message of the Envelope. In case of symmetric encryption, it also contains appended Salt (a.k.a. AES Nonce, 12 bytes). Plaintext (unencrypted) payload consists of the following concatenated fields: flags, auxiliary field, payload, padding and signature (in this sequence).
+
+	flags: 1 byte; first two bits contain the size of auxiliary field, third bit indicates whether the signature is present.
+	
+	auxiliary field: up to 4 bytes; contains the size of payload.
+
+	payload: byte array of arbitrary size (may be zero).
+
+	padding: byte array of arbitrary size (may be zero).
+
+	signature: 65 bytes, if present.
+
+	salt: 12 bytes, if present (in case of symmetric encryption).
+
+Those unable to decrypt the message data are also unable to access the signature. The signature, if provided, is the ECDSA signature of the Keccak-256 hash of the unencrypted data using the secret key of the originator identity. The signature is serialised as the concatenation of the `R`, `S` and `V` parameters of the SECP-256k1 ECDSA signature, in that order. `R` and `S` are both big-endian encoded, fixed-width 256-bit unsigned. `V` is an 8-bit big-endian encoded, non-normalised and should be either 27 or 28.
+
+The padding field was introduced in order to align the message size, since message size alone might reveal important metainformation. Padding can be arbitrary size. However, it is recommended that the size of Data Field (excuding the Salt) before encryption (i.e. plain text) should be factor of 256 bytes.
+
+### Payload Encryption
+
+Asymmetric encryption uses the standard Elliptic Curve Integrated Encryption Scheme with SECP-256k1 public key.
+
+Symmetric encryption uses AES GCM algorithm with random 96-bit nonce.
+
+## Additional capabilities
+
+Waku supports multiple capabilities. These include light node, rate limting, mailserver (client and server) and bridging of traffic. Here we list these capabilities, how they are identified, what properties they have and what invariants they must maintain.
+
+### Light node
+
+The rationale for light nodes is to allow for interaction with waku on resource restricted devices as bandwidth can often be an issue.
+
+Light nodes MUST NOT forward any incoming messages, they MUST only send their own messages. Light nodes SHOULD not connect to other light nodes, this would result in messages being dropped between the two.
```

Comment:
```suggestion
Light nodes MUST NOT forward any incoming messages, they MUST only send their own messages. When light nodes happen to connect to each other, they SHOULD disconnect. As this would result in messages being dropped between the two.
```
It kinda depends on the discovery mechanism whether they would ever connect to each other or not. But if e.g. discv4 is still used this would be possible. Hence the suggestion.